lib/doorkeeper/oauth/pre_authorization.rb

```diff
@@ -5,14 +5,16 @@ module OAuth
     class PreAuthorization
       include Validations
 
-      validate :response_type, error: :unsupported_response_type
       validate :client, error: :invalid_client
-      validate :scopes, error: :invalid_scope
       validate :redirect_uri, error: :invalid_redirect_uri
+      validate :params, error: :invalid_request
```

Comment:
Change the order of `validation` to make sure that `AuthorizationsController` response properly when error happens, check that: `Does that ErrorResponse redirectable?`

- Because with missing or invalid `client` and `redirect_uri` param, `AuthorizationsController` must not redirect client to `redirect_uri`, that is checked at [here](https://github.com/doorkeeper-gem/doorkeeper/blob/99f83f0dd88a215eb5b5f815f4b2a9e07caf9b5d/lib/doorkeeper/oauth/error_response.rb#L42) and called at [here](https://github.com/doorkeeper-gem/doorkeeper/blob/99f83f0dd88a215eb5b5f815f4b2a9e07caf9b5d/app/controllers/doorkeeper/authorizations_controller.rb#L54). So this order make sure that we have checked `client` and `redirect_uri` first.